py/torch_tensorrt/dynamo/lowering/passes/nodeclassifier.py

```diff
@@ -0,0 +1,300 @@
+
+    def _check_inner(self, node):
+        return any(
+            re.match(regex, node.name) for regex in self.disabled_node_name_regex
+        )
+
+
+class DisabledTargets(NodeRuleBase):
+    """Rule for keeping nodes with specific operation types in high precision."""
+
+    def __init__(self, targets_to_exclude):
+        """Initialize the rule.
+
+        Args:
+            targets_to_exclude: List of operation types to keep in high precision.
+        """
+        self.targets_to_exclude = targets_to_exclude
+
+    def _check_inner(self, node):
+        return node.target in self.targets_to_exclude
+
+
+class IORangeRule(NodeRuleBase):
+    """Rule for keeping nodes with out-of-range inputs/outputs in high precision."""
+
+    def __init__(self, data_max, reference_data):
+        """Initialize the rule.
+
+        Args:
+            data_max: Maximum absolute value allowed for node I/O.
+            reference_data: Reference data for checking I/O ranges.
+        """
+        self.data_max = data_max
+        self.reference_data = reference_data
+        self.output_data = None
+
+    def _check_inner(self, node):
+        def is_io_out_of_range(node):
+            tensor_name = node.name
+            if tensor_name not in self.reference_data:
+                logger.debug(
+                    f"Node {node.name}: Tensor {tensor_name} not found in reference data. Skipping I/O range check."
+                )
+                return False
+            ref_data = self.reference_data[tensor_name]
+            if ref_data.numel() == 0:
+                logger.debug(
+                    f"Node {node.name}: Tensor {tensor_name} has 0 elements. Skipping I/O range check."
+                )
+                return False
+            logger.debug(
+                f"Node {node.name}: reference data: min={ref_data.min()}, max={ref_data.max()}"
+            )
+            if torch.any(torch.abs(ref_data) > self.data_max):
+                self.output_data = ref_data
+                return True
+
+        if self.reference_data:
+            for in_node in node.all_input_nodes:
+                if is_io_out_of_range(in_node):
+                    return True
+            for out_node in list(node.users):
+                if is_io_out_of_range(out_node):
+                    return True
+        return False
+
+    def _log_skipped(self, node, **kwargs):
+        """Log information about skipped nodes with I/O range violations."""
+        if self.output_data is not None:
+            logger.info(
+                f"Skipping node {node.name}: reference IO out of range: min={torch.min(self.output_data)}, "
+                f"max={torch.max(self.output_data)}, range=[{-self.data_max}, {self.data_max}]"
+            )
+        else:
+            super()._log_skipped(node, **kwargs)
+
+
+class DepthOfReductionRule(NodeRuleBase):
+    """Rule for keeping nodes with high depth of reduction in high precision."""
```

Comment:
could you add more detailed information about this class with an example in docstring ?